sharding/src/main/scala/com/lightbend/akka/bench/sharding/scalability/ActorCountingBenchmarkMaster.scala

```diff
@@ -26,54 +26,60 @@ import akka.stream.scaladsl.{ Flow, Source }
 import com.lightbend.akka.bench.sharding.BenchSettings
 
 import scala.concurrent.duration._
+import akka.actor.Props
 
-class ActorCountingBenchmarkMaster extends Actor with ActorLogging {
+object ActorCountingBenchmarkMaster {
+  def props(sharding: ActorRef): Props =
+    Props(new ActorCountingBenchmarkMaster(sharding))
+}
+
+class ActorCountingBenchmarkMaster(sharding: ActorRef) extends Actor with ActorLogging {
   implicit val dispatcher = context.system.dispatcher
   val settings = BenchSettings(context.system)
   val addActorsBatch: Int = settings.AddActorsPerBatch
   val addActorsInterval: FiniteDuration = settings.AddActorsInterval
-  
+
   val cluster = Cluster(context.system)
-  
+
   // number of actors we sent a message to (so they should start in sharding)
   var totalStartingActorsInSharding = 0L
   // number of actors who have not yet replied back that they've started
   var pendingAliveConfirmation = 0L
   var totalAliveConfirmedActors = 0L
-  val timesWhenWeStartedBatch = new util.ArrayDeque[Long](10) 
-  
+  var batchCount = 0
+  var timesWhenWeStartedBatch = Map.empty[Int, Long]
```

Comment:
Hm... lost which one, maybe I need to open the code in my ide to get that, but if it's quick to explain please do :)